New CI builds

Ham22 opened the pull request:
This connects to #48 and aims to remove the need for https://github.com/CreatorDev/ci repo entirely. Instead the CI system can clone this repo directly and run the build based on the Jenkinsfile.

For now any CI 'hacks' have been moved here so they are all in one place. Will migrate as and when needed.

Ham22 committed:
ci40: configure branch feeds for platform only

This branch is dedicated to only getting the ci40 platform working
and nothing to do with the creator/kit feeds so don't add them
here.

Note this change lives here and will never be accepted upstream.
Once all of the ci40 board support is upstreamed this change can
be removed.

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>

Ham22 commented:
Note that this is an RFC and not expected to work/be merged yet

abhijit-mahajani-imgtec reviewed (approved):
Looks good to me. however mainly needs verification to see how should patches need to be handled via forked packages repo

nikhil-zinjurde-imgtec commented:
build all

nikhil-zinjurde-imgtec commented on scripts/build.sh:
```diff
@@ -0,0 +1,80 @@
+	echo "# (Ignoring errors for any package built as a module!)"
+	BUILD_OPTIONS="IGNORE_ERRORS=m"
+fi
+make $BUILD_OPTIONS -j $THREADS
```
need V=s either permanently in here or as a usage option

Ham22 replied:
Migrated to Jenkinsfile and have added some magic build logic for this.

nikhil-zinjurde-imgtec commented:
build all

Ham22 committed:
Add initial Jenkinsfile

With this in the repo and Jenkins pipelines configured correctly we
are able to control the CI builds from the repo itself meaning it is
nicely tracked per branch in the tree history.

For now all CI 'hacks' have been moved here. Can migrate as needed.

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>

Ham22 commented:
The new CI system generates some very long paths which has exposed a mkimage bug. Fix has been applied in u-boot but need to update the feed for it to work https://github.com/CreatorDev/Ci40-platform-feed/pull/97

Ham22 committed:
mkimage: add patch to increase dtc cmd buffer

Previously any builds with long paths could cause the buffer to be
truncated causing undefined behaviour. The patch increases the
buffer and adds some limits checking.

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>
